src/components/Forms/CreatePostForm/CreatePostForm.js

```diff
@@ -0,0 +1,134 @@
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import cookie from 'react-cookies';
+import TextField from '@material-ui/core/TextField';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogContentText from '@material-ui/core/DialogContentText';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import config from '../../../config';
+import { serviceRequest } from '../../../services/serviceRequest';
+import { ErrorMessage } from '../../ErrorMessage';
+
+const path = '/api/v1/user/createpost';
+const domain = config.apiDomain;
+
+function preparePayload(method, data) {
+  const token = cookie.load('jwtToken');
+  const requestHeader = { Authorization: token };
+  const url = `https://${domain}${path}`;
+  return {
+    method,
+    url,
+    headers: requestHeader,
+    data,
+  };
+}
+
+export const CreatePostForm = () => {
+  const [open, setOpen] = useState(false);
+  const [formData, setForm] = useState({ label: 'general', title: null, content: null });
+  const [isError, setIsError] = useState(false);
+  const [errorMsg, setErrorMsg] = useState('Please make sure all fields are complete');
+
+  const handleChange = (event) => {
+    const updateForm = { ...formData };
+    updateForm[event.target.name] = event.target.value;
+    setForm(updateForm);
+  };
+
+  const handleClickOpen = () => {
+    setOpen(true);
+  };
+
+  const handleClose = () => {
+    setOpen(false);
+    setErrorMsg('');
+  };
+
+  const handleCreate = async () => {
+    if (!formData.label || !formData.title || !formData.content) {
+      setErrorMsg('Please make sure all fields are complete');
+      setIsError(true);
+      return;
+    }
```

Comment:
Don't think this is necessary because the <Textfield> with property `required` checked this

Reply:
At first, I thought having `required` would be good enough. However, with `<TextField/>` `required` does not prevent user from clicking `CREATE` button. So a bad request will be sent to the server. I am trying to use this block of code to prevent that.

Reply:
if add a `<form onSubmit={handleCreate}>` to surround `textfield` and `button`, and change the type of `button` to submit, then the required will work

Reply:
I will try that